### ISSUE TITLE
Explore the DDPM pytorch repo

### ISSUE THREAD

**Issue by mbaddar1** [mbaddar1/genmodel#10]
https://github.com/mbaddar1/denoising-diffusion-pytorch
Understand 

- UNet architecture
- Loss function
- The optimization
- Try to adapt for simple cases like swissroll and conc circle
- Understand the FID score
